model_access_restriction/views/ir_model_access_restriction_views.xml

```diff
@@ -0,0 +1,136 @@
+                        <!-- <field name="perm_read"/>
+                        <field name="perm_write"/> -->
+                        <field name="perm_create" />
+                        <field name="perm_unlink" />
+                    </group>
+                </group>
+                <group string="Allowed Groups">
+                    <field name="groups" nolabel="1" colspan="4" />
+                </group>
+                <i
+                        class="fa fa-info fa-3x text-info float-left"
+                        role="img"
+                        aria-label="Info"
+                        title="Info"
+                    />
+                <h3>Interaction between access records</h3>
+                <div>
+                    <p>
+                        Normal access records give permissions. Access restriction records removes permissions, if a user does not have one group of a resctriction the access will be forbidden.
+                    </p>
+                    Detailed algorithm:
+                    <ol>
+                        <li
+                            >Normal access records are combined together with a logical OR operator, if a user has the group of an acess record access, access will be granted</li>
+                        <li
+                            >Access restriction records are applied before normal access, and combines together with a logical AND operator. If a restriction is not met, the access is forbidden.</li>
+                        <li
+                            >A restriction is not met if the user does not belong to any of the allowed groups.</li>
+                    </ol>
+                    <p
+                        >Example: ACCESS_1 OR ACCESS_2 AND (ACCESS_RESTRICTION_1 AND ACCESS_RESTRICTION_2)
+                        <ol>
+                            <li>ACCESS_1. Group: Internal Users</li>
+                            <li>ACCESS_2. Group: Administrator</li>
+                            <li
+                                >ACCESS_RESTRICTION_1. Allowed Groups: Internal Users</li>
+                            <li>ACCESS_RESTRICTION_2. Allowed Groups: Administrator</li>
+                        </ol>
+                        A internal user won't have access in this example because it does not met the ACCESS_RESTRICTION_2. TRUE OR FALSE AND (TRUE AND FALSE) = TRUE AND (FALSE) = FALSE
+                    </p>
+                </div>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record id="view_model_access_restrictions_tree" model="ir.ui.view">
+        <field name="model">ir.model.access.restriction</field>
+        <field name="arch" type="xml">
+            <tree decoration-info="not groups">
+                <field name="name" />
+                <field name="model_id" />
+                <field
+                    name="groups"
+                    widget="many2many_tags"
+                    options="{'no_create':True}"
```

Comment:
why not readonly in the model?

Reply:
This field is not readonly and it don't has to be. Its the allowed permission groups that the users have to fill